_automating-workflows/workflow-tutorial.md

```diff
@@ -0,0 +1,613 @@
+---
+layout: default
+title: Workflow tutorial
+nav_order: 20
+---
+
+# Workflow tutorial
+
+You can automate the setup of common use cases, such as conversational chat, using a Chain-of-Thought (CoT) agent. This page presents a complete example of setting up a CoT agent.
+
+The setup requires the following sequence of API requests, with provisioned resources used in subsequent requests. The following list is an overview of the steps required for this workflow. The step names correspond to the names in the template:
+
+1. **Deploy a model on the cluster**
+    * [Step `create_connector_1`](#step-create_connector_1): Create a connector to an externally hosted model
+    * [Step `register_model_2`](#step-register_model_2): Register a model using the connector that you created
+    * [Step `deploy_model_3`](#step-deploy_model_3): Deploy the model
+1. **Use the deployed model for inference**
+    * Set up several tools which perform specific tasks:
+      * [Step `math_tool`](#step-math_tool): Set up a math tool
+      * [Step `ml_model_tool`](#step-ml_model_tool): Set up a machine learning (ML) model tool
+    * Set up one or more agents which use some combination of the tools:
+      * [Step `sub_agent`](#step-sub_agent): Create an agent that uses the math tool
+    * Set up tools representing these agents:
+      * [Step `agent_tool`](#step-agent_tool): Wrap the `sub_agent` so you can use it as a tool
+    * [Step `root_agent`](#step-root_agent): Set up a root agent which may delegate the task to either tools or another agent
+
+The following sections describe the steps in detail. For the complete template of these steps, see [Complete YAML workflow template](#complete-yaml-workflow-template).
+
+## Workflow graph
+
+The workflow described in the previous section is organized into a [template](#complete-yaml-workflow-template). Note that you can order the steps in several ways. In the example template, the `ml_model_tool` step is specified right before the `root_agent` step, but you can specify it at any point after the `deploy_model_3` step and before the `root_agent` step. The following diagram shows the directed acyclic graph (DAG) that OpenSearch creates for all of the steps in the order specified in the template.
+
+![Example workflow steps graph]({{site.url}}{{site.baseurl}}/images/automatic-workflow-dag.png){:style="width: 100%; max-width: 600px;" class="img-centered"}
+
+## 1. Deploy a model on the cluster
+
+To deploy a model on the cluster, you need to create a connector to the model, register the model, and deploy the model.
+
+### Step `create_connector_1`
```

Comment:
Global: The order of words here is a little confusing (and in the list above). I can see why you did it this way, but I'm not a fan.